### DISCUSSION TITLE
Storing different Images in a Dictionary

### DISCUSSION THREAD

**Discussion by PPS100** [emgucv/emgucv#533] [Q&A]
I have a Dictionary as outlined below. I am trying to create a Dictionary that will how different types of images. Example are below.

Dictionary:
protected Dictionary<string, Image<Emgu.CV.IColor, byte>> _Images = new Dictionary<string, Image<Emgu.CV.IColor, byte>>();

Example Usage of Dictionary
 Image<Bgr, Byte> templateOverlay = new Image<Bgr, Byte>(3.Size);
_Images.Add("Test1", templateOverlay )

Image<Gray, Byte> contourImage = new Image<Gray, byte>(3)
_Images.Add("Test2", contourImage )

**Comment by emgucv** [comment]
It is recommended to use Emgu.CV.Mat (cv::Mat)  instead of Image<,> (IplImage).

Open CV is moving away from IplImage since v2.x, please migrate to Emgu.CV.Mat (cv::Mat) for future compatibility. 

If using Image<,> is a requirement, please create the dictionary as Dictionary<string, IInputArray>. IInputArray is a based class of Image<,>.

**Reply by PPS100** [reply]
Perfect....Thank You for this guidance.
